implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -163,13 +156,29 @@ private ConfigSourceInterceptorContext buildInterceptorChain(final SmallRyeConfi
                     return null;
                 }, null);
 
-        for (int i = interceptors.size() - 1; i >= 0; i--) {
-            current = new SmallRyeConfigSourceInterceptorContext(interceptors.get(i), current);
+        List<ConfigSourceInterceptor> configSourceInterceptors = initInterceptors(interceptors, current);
+        for (int i = configSourceInterceptors.size() - 1; i >= 0; i--) {
+            current = new SmallRyeConfigSourceInterceptorContext(configSourceInterceptors.get(i), current);
```

Comment:
Why recreate the contexts here?  You could reuse the same context instances and do the init in a single loop.

Reply:
I guess I can return the context instances instead. The idea was to avoid reusing the context passed to the factory, since it wouldn't be the final chain.

Reply:
Another reason why I've done it that way is because it is easier to build the chain from backwards to the first, since we want our last interceptor to be the one that actual lookups the value, so we can link them together to the last one by least priority to highest. Start by `lookup` then `lowest->lookup`, `highest->lowest->lookup`.

On constructor, we want to initialize them on priority as well, so it means that we need to build the chain `highest->lookup`, meaning that we would need to break the chain in the middle to insert less priority interceptors `highest->low->lookup`

Do you thing we should change this?

Reply:
I think we should change it so that interceptors are initialized in last-to-first order.  That makes everything consistent and it means that an interceptor will never "see" a subsequent interceptor on initialization that it wouldn't normally "see".

Reply:
IOW build it like this:

* `lookup`
* `ctxt->lookup`
* `interceptor(last)->ctxt->lookup`
* `ctxt->interceptor(last)->ctxt->lookup`

etc., initializing at the same time that it is built.

Reply:
Sure.